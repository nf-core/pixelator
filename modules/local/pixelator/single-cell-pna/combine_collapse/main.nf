process PIXELATOR_PNA_COMBINE_COLLAPSE {
    tag "${meta.id}"
    label 'process_medium'

    // TODO: Add conda
    // conda "bioconda::pixelator=0.18.2"

    container "${workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container
        ? 'quay.io/pixelgen-technologies/pixelator:0.21.2'
        : 'quay.io/pixelgen-technologies/pixelator:0.21.2'}"

    input:
    tuple val(meta), path(parquet_files, stageAs: "parquet/*"), path(json_report_files, stageAs: "reports/*")

    output:
    tuple val(meta), path("collapse/*.parquet", arity: 1),    emit: parquet
    tuple val(meta), path("collapse/*.report.json"),          emit: report_json
    tuple val(meta), path("collapse/*.meta.json"),            emit: metadata_json
    tuple val(meta), path("*pixelator-combine-collapse.log"), emit: log

    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    // --design is passed in meta and added to args through modules.conf
    prefix = task.ext.prefix ?: "${meta.id}"
    def args = task.ext.args ?: ''
    def parquet_args = parquet_files.join(' --parquet ')
    def report_args = json_report_files.join(' --report ')

    """
    pixelator \\
        --log-file ${prefix}.pixelator-combine-collapse.log \\
        --verbose \\
        single-cell-pna \\
        combine-collapse \\
        --output . \\
        ${args} \\
        --parquet ${parquet_args} \\
        --report ${report_args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        pixelator: \$(echo \$(pixelator --version 2>/dev/null) | sed 's/pixelator, version //g' )
    END_VERSIONS
    """

    stub:
    prefix = task.ext.prefix ?: "${meta.id}"

    """
    mkdir collapse
    touch collapse/${prefix}.collapse.parquet
    touch collapse/${prefix}.collapse.meta.json
    touch collapse/${prefix}.report.json
    touch ${prefix}.pixelator-combine-collapse.log

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        pixelator: \$(echo \$(pixelator --version 2>/dev/null) | sed 's/pixelator, version //g' )
    END_VERSIONS
    """
}
