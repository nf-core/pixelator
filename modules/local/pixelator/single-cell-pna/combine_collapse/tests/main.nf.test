nextflow_process {

    name "Test Process PIXELATOR_PNA_COMBINE_COLLAPSE"
    script "../main.nf"
    process "PIXELATOR_PNA_COMBINE_COLLAPSE"
    tag "modules"
    tag "pixelator"
    tag "pixelator/pna"
    tag "modules_pixelator"
    tag "pixelator/single_cell_pna_combine_collapse"


    test("PNA combine collapse - multiple files") {
        when {
            process {
                """
                input[0] = [
                    [ id:'PNA055_Sample07_filtered_S7', design:'pna-2', panel:'proxiome-immuno-155', technology:'pna' ],
                    [
                        file(params.pipelines_testdata_base_path + 'testdata/pna/modules/collapse/PNA055_Sample07_filtered_S7.collapse.m1.part_000.parquet', checkIfExists: true),
                        file(params.pipelines_testdata_base_path + 'testdata/pna/modules/collapse/PNA055_Sample07_filtered_S7.collapse.m1.part_001.parquet', checkIfExists: true),
                        file(params.pipelines_testdata_base_path + 'testdata/pna/modules/collapse/PNA055_Sample07_filtered_S7.collapse.m2.part_000.parquet', checkIfExists: true),
                        file(params.pipelines_testdata_base_path + 'testdata/pna/modules/collapse/PNA055_Sample07_filtered_S7.collapse.m2.part_001.parquet', checkIfExists: true),
                    ],
                    [
                        file(params.pipelines_testdata_base_path + 'testdata/pna/modules/collapse/PNA055_Sample07_filtered_S7.collapse.m1.part_000.report.json', checkIfExists: true),
                        file(params.pipelines_testdata_base_path + 'testdata/pna/modules/collapse/PNA055_Sample07_filtered_S7.collapse.m1.part_001.report.json', checkIfExists: true),
                        file(params.pipelines_testdata_base_path + 'testdata/pna/modules/collapse/PNA055_Sample07_filtered_S7.collapse.m2.part_000.report.json', checkIfExists: true),
                        file(params.pipelines_testdata_base_path + 'testdata/pna/modules/collapse/PNA055_Sample07_filtered_S7.collapse.m2.part_001.report.json', checkIfExists: true)
                    ]
                ]
                """
            }
        }


        then {
            assertAll(
                { assert process.success },
                {
                    with(process.out.parquet) {
                        assert path(get(0).get(1)).exists()
                    }
                },
                { assert snapshot(
                        process.out.report_json,
                        process.out.metadata_json,
                        process.out.versions
                    ).match()
                },
                {
                    with (process.out.log) {
                        assert path(get(0).get(1)).readLines().last().contains("Finished pixelator combine-collapse")
                    }
                },
            )
        }
    }
}
