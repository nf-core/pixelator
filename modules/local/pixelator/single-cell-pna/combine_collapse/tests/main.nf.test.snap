{
    "PNA combine collapse - multiple files": {
        "content": [
            [
                [
                    {
                        "id": "PNA055_Sample07_filtered_S7",
                        "design": "pna-2",
                        "panel": "proxiome-immuno-155",
                        "technology": "pna"
                    },
                    "PNA055_Sample07_filtered_S7.report.json:md5,f072efda6e9a17670d903151390b4c40"
                ]
            ],
            [
                [
                    {
                        "id": "PNA055_Sample07_filtered_S7",
                        "design": "pna-2",
                        "panel": "proxiome-immuno-155",
                        "technology": "pna"
                    },
                    "PNA055_Sample07_filtered_S7.meta.json:md5,b5dec890f6bc2cb309626e9db9d9bad9"
                ]
            ],
            [
                "versions.yml:md5,5b7f60bcecd4a25456148724b21ae96e"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-26T11:59:39.205308002"
    }
}